stable/graylog/templates/statefulset.yaml

```diff
@@ -0,0 +1,248 @@
+              {{- end }}
+              {{- end }}
+              {{- if .Values.graylog.metrics.enabled }}
+                echo "Downloading https://github.com/graylog-labs/graylog-plugin-metrics-reporter/releases/download/2.4.0-beta.3/metrics-reporter-prometheus-2.4.0-beta.3.jar ..."
+                curl -s --location --retry 3 -o ${GRAYLOG_HOME}/plugin/metrics-reporter-prometheus-2.4.0-beta.3.jar "https://github.com/graylog-labs/graylog-plugin-metrics-reporter/releases/download/2.4.0-beta.3/metrics-reporter-prometheus-2.4.0-beta.3.jar"
+              {{- end }}
+                chown -R ${GRAYLOG_USER}:${GRAYLOG_GROUP} ${GRAYLOG_HOME}/plugin/
+              {{- if .Values.graylog.geoip.enabled }}
+                echo "Downloading Maxmind GeoLite2 ..."
+                curl -s --location --retry 3 -o /tmp/GeoLite2-City.tar.gz "https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.tar.gz"
+                curlreturn=$?
+                if [[ $curlreturn -eq 0 ]]; then
+                  mkdir -p ${GRAYLOG_HOME}/geoip && cd ${GRAYLOG_HOME}/geoip && tar xvzf /tmp/GeoLite2-City.tar.gz --wildcards "*.mmdb" --strip-components=1 -C ${GRAYLOG_HOME}/geoip && chown -R graylog:graylog ${GRAYLOG_HOME}/geoip
+                fi
+              {{- end }}
+                # Start Graylog
+                echo "Starting graylog"
+                # Original docker-entrypoint.sh in Graylog Docker will error while executing since you can't chown readonly files in `config`
+                # exec /docker-entrypoint.sh graylog
+                echo "Graylog Home ${GRAYLOG_HOME}"
+                echo "Graylog User ${GRAYLOG_USER}"
+                echo "JVM Options ${GRAYLOG_SERVER_JAVA_OPTS}"
+                "${JAVA_HOME}/bin/java" \
+                  ${GRAYLOG_SERVER_JAVA_OPTS} \
+                  -jar \
+                  -Dlog4j.configurationFile=${GRAYLOG_HOME}/config/log4j2.xml \
+                  -Djava.library.path=${GRAYLOG_HOME}/lib/sigar/ \
+                  -Dgraylog2.installation_source=docker \
+                  ${GRAYLOG_HOME}/graylog.jar \
+                  server \
+                  -f ${GRAYLOG_HOME}/config/graylog.conf
+          env:
+            - name: GRAYLOG_SERVER_JAVA_OPTS
+              {{- $javaOpts := "-Djava.net.preferIPv4Stack=true -XX:NewRatio=1 -server -XX:+ResizeTLAB -XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled -XX:+UseParNewGC -XX:-OmitStackTraceInFastThrow" }}
+              {{- if .Values.graylog.heapSize -}}
+              value: "{{ $javaOpts }} {{ printf "-Xms%s -Xmx%s" .Values.graylog.heapSize .Values.graylog.heapSize}}"
+              {{- else if .Values.graylog.resources.requests.memory -}}
+              {{ $heapSize := regexFind "[\\d]+" .Values.graylog.resources.requests.memory}}
+              value: "{{ $javaOpts }} {{ printf "-Xms%sm -Xmx%sm" $heapSize $heapSize }}"
+              {{- else -}}
```

Comment:
@AntonioMeireles I updated chart to use `-XX:+UseCGroupMemoryLimitForHeap` when no `heapsize` specified. 

For calculating heap size from `requests.memory`, it does not have a problem since it was calculated from `resources.requests`. The container memory limited from `resources.limits` 
But this method is contradict with `-XX:+UseCGroupMemoryLimitForHeap` so I remove it anyway.

Thanks for pointing out this JVM option.

Note: from my testing `-XX:+UseCGroupMemoryLimitForHeap` will make Graylog initializing slower than specifying the `heapSize` since its start with very low heap size.